plugins/automatic_alignment.py

```diff
@@ -0,0 +1,367 @@
+# -*- coding: utf-8 -*-
+'''
+Created on 10 April 2017
+
+@author: Guilherme Stiebler
+
+Gives ability to automatically change the overlay-metadata.
+
+Copyright © 2017 Éric Piel, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms of the GNU
+General Public License version 2 as published by the Free Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis. If not,
+see http://www.gnu.org/licenses/.
+'''
+
+from __future__ import division
+
+from collections import OrderedDict
+import functools
+import logging
+import math
+from odemis import dataio, model
+from odemis.acq import stream
+from odemis.gui.plugin import Plugin, AcquisitionDialog
+from odemis.util.dataio import data_to_static_streams, open_acquisition
+from odemis.acq.align import keypoint
+from odemis.util.conversion import get_img_transformation_md
+import odemis.gui.util as guiutil
+from odemis.gui.conf import get_acqui_conf
+from scipy import ndimage
+import os
+import wx
+import cv2
+from odemis.gui.util import call_in_wx_main
+from odemis.util import img
+import numpy
+
+class AlignmentAcquisitionDialog(AcquisitionDialog):
+
+    @call_in_wx_main
+    def addStream(self, stream, index):
+        """
+        Adds a stream to the canvas, and a stream entry to the stream panel.
+        It also ensures the panel box and canvas are shown.
+        """
+        new_stream_l = index == 0 and not self.viewport_l.IsShown()
+        new_stream_r = index == 1 and not self.viewport_r.IsShown()
+
+        if index == 0:
+            self.viewport_l.Show()
+        else:
+            self.viewport_r.Show()
+
+        if new_stream_l or new_stream_r:
+            self.Layout()
+            self.Fit()
+            self.Update()
+
+        if stream:
+            if index == 0:
+                self.streambar_controller.addStream(stream)
```

Comment:
Never add the stream to the strembar_controller. No stream entry should be displayed anyway.